shared/src/popup.js

```diff
@@ -351,6 +260,87 @@ async function setup() {
     }
   });
 
+  async function handleGetData({
+    token,
+    api_token,
+    sync_existing,
+    api_engine,
+  } = {}) {
+    if (token && eStatus) {
+      eStatus.classList.remove('status_error');
+      eStatus.classList.add('status_good');
+
+      eTokenInput.value = token;
+      eApiTokenInput.value = api_token;
+
+      if (sync_existing) {
+        setStatus('auto_token');
+      } else {
+        setStatus('manual_token');
+      }
+
+      if (api_token) {
+        eApiParams.forEach((element) => {
+          element.style.display = '';
+        });
+      }
+
+      if (api_engine) {
+        eApiEngineSelect.value = api_engine;
+      }
+
+      const hasIncognitoAccess =
+        await browser.extension.isAllowedIncognitoAccess();
+
+      if (!hasIncognitoAccess) {
+        const eIncognito = document.querySelector('#incognito');
+
+        if (!eIncognito) {
+          console.error('No incognito div to place text');
+          return;
+        }
+
+        eIncognito.style.display = '';
+
+        const eFirefoxExt = document.querySelector('#firefox_ext');
+        const eChromeExt = document.querySelector('#chrome_ext');
+
+        if (!IS_CHROME) {
+          eFirefoxExt.style.display = '';
+          eChromeExt.style.display = 'none';
+        } else {
+          eFirefoxExt.style.display = 'none';
+          eChromeExt.style.display = '';
+        }
+
+        // NOTE: slight little hack to make the chrome://extensions link not be blocked.
+        if (IS_CHROME) {
+          const eChromeLink = document.querySelector('#chrome_link');
+          if (eChromeLink) {
+            eChromeLink.addEventListener('click', async () => {
+              await browser.runtime.sendMessage({ type: 'open_extension' });
+            });
+          }
+        }
+      }
+    }
+  }
+
+  const sessionObject = await browser.storage.local.get('session_token');
+  const syncObject = await browser.storage.local.get('sync_existing');
+  const apiObject = await browser.storage.local.get('api_token');
+  const apiEngineObject = await browser.storage.local.get('api_engine');
+
+  await handleGetData({
+    token: sessionObject?.session_token,
+    sync_existing:
+      typeof syncObject?.sync_existing !== 'undefined'
+        ? syncObject.sync_existing
+        : true,
+    api_token: apiObject?.api_token,
+    api_engine: apiEngineObject?.api_engine,
+  });
```

Comment:
Here I removed the call to the service worker to get the data as that could sometimes fail, while this doesn't. This is also a tidbit faster.